google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/SessionPoolTest.java

```diff
@@ -140,6 +140,31 @@ public void sessionCreation() {
     }
   }
 
+  @Test
+  public void poolLifo() {
+    setupMockSessionCreation();
+    pool = createPool();
+    Session session1 = pool.getReadSession();
+    Session session2 = pool.getReadSession();
+    assertThat(session1).isNotEqualTo(session2);
+
+    session2.close();
+    session1.close();
+    Session session3 = pool.getReadSession();
+    Session session4 = pool.getReadSession();
+    assertThat(session3).isEqualTo(session1);
+    assertThat(session4).isEqualTo(session2);
+    session3.close();
+    session4.close();
+
+    Session rw1 = pool.getReadWriteSession();
+    Session rw2 = pool.getReadWriteSession();
```

Comment:
[nit] Naming wise, I would call these session5 and session6 to be consistent with the variable naming elsewhere in this test and file, and since the type is not relevant.

Reply:
Renamed to session5 and session6.